codex/purchasing.nim

```diff
@@ -77,11 +76,8 @@ proc run(purchase: Purchase) {.async.} =
       done.complete()
     let request = purchase.request
     let subscription = await market.subscribeFulfillment(request.id, callback)
-    try:
-      await done
-      purchase.selected = await market.getHost(request.id)
-    finally:
-      await subscription.unsubscribe()
+    await done
```

Comment:
what if `done` throws, try/finally seems safer here?

Reply:
`done` is a variable local to this proc, that is only completed on line 76. It won't throw.